libraries/AP_RangeFinder/AP_RangeFinder_TeraRangerDUO.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include "AP_RangeFinder_TeraRangerDUO.h"
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_Math/crc.h>
+#include <GCS_MAVLink/GCS.h>
+#include <ctype.h>
+#include <stdio.h>
+
+/* 
+   The constructor also initialises the rangefinder. Note that this
+   constructor is not called until detect() returns true, so we
+   already know that we should setup the rangefinder
+*/
+AP_RangeFinder_TeraRangerDUO::AP_RangeFinder_TeraRangerDUO(RangeFinder::RangeFinder_State &_state,
```

Comment:
I am told that there is a class of teraranger I2C.
This DUO is a UART.
I think the TeraRangerSerial class.

Reply:
Agree.